snuba/query/snql/parser.py

```diff
@@ -1300,23 +1301,28 @@ def _post_process(
 def parse_snql_query(
     body: str, dataset: Dataset, custom_processing: Optional[CustomProcessors] = None,
 ) -> Union[CompositeQuery[QueryEntity], LogicalQuery]:
-    query = parse_snql_query_initial(body)
+    with sentry_sdk.start_span(op="parser", description="parse_snql_query_initial"):
+        query = parse_snql_query_initial(body)
 
-    _post_process(
-        query, POST_PROCESSORS,
-    )
+    with sentry_sdk.start_span(op="processor", description="post_processors"):
+        _post_process(
+            query, POST_PROCESSORS,
```

Comment:
I would add a span for each processor as well. You could probably do it programatically in the `_post_process` function doing `start_span(... description=func.__name__)`. If you add it to `_post_process` it will also have the benefit of capturing all the other post processing functions below.